solana/caip19.md

```diff
@@ -0,0 +1,57 @@
+---
+namespace-identifier: solana-caip19
+title: Solana Namespace - Asset
+author: Qiao Liang (@qbig)
+discussions-to: 
+status: Draft
+type: Standard
+created: 2022-06-01
+updated: 2022-06-01
+requires: ["CAIP-2", "CAIP-10", "CAIP-19", "CAIP-30"]
+---
+
+# CAIP-19
+
+*For context, see the [CAIP-19][] specification.*
+
+## Rationale
+
+On solana, all contracts are stateless and state is instead represented using "accounts". More specifically, all fungible and non-fungible tokens are using the same instance of token contracts and for each token:
+
+- 1 single global "mint" account instance is used for the global data like "total supply".
+- each token account represent an account balance belong to a certain address.
+
+And the difference between fungible token and non-fungible token is that non-fungible token mint has a total supply of 1 and zero decimal place.
+But both asset type can be identified by their "mint" account. 
+
+
+## Syntax
+
+After the CAIP-2 namespace+chainID, a slash defines an `asset_namespace` and an `asset_reference`. And since both fungible and non-fungible tokens can be identified using the mint account, we could use `token` as the namespace and the address of the mint account as the reference.
```

Comment:
How about Metaplex [Token Metadata v1.1](https://docs.metaplex.com/programs/token-metadata/changelog/v1.1) which now includes [Token Standard Field](https://docs.metaplex.com/programs/token-metadata/token-standard#the-token-standard-field)?
In this sense we could use:
```ts
// Equivalent to ERC20
'solana:4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZ/Fungible:EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v'

// Equivalent to ERC721
'solana:4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZ/NonFungible:Fz6LxeUg5qjesYX3BdmtTwyyzBtMxk644XiTqU5W3w9w'

// Equivalent to ERC1155
'solana:4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZ/NonFungibleEdition:Fz6LxeUg5qjesYX3BdmtTwyyzBtMxk644XiTqU5W3w9w'
```

Reply:
@felipe-chamas it's a great suggestion as it indeed implies more information. but two issues with Token Standard Field is that they are:
1. long
2. implied rather than speficied directly (if this and this, then it's NonFungibleEdition)

So how about just `token` and `nft`, `edition` ?

Reply:
@bumblefudge @ligi tht?

Reply:
yes i prefer the abbreviations, maybe just add a little table like

|reference|equivalent to|see also|
|---|---|---|
| `token` | native tokens, equivalent | [Metaplex](https://docs.metaplex.com/programs/token-metadata/token-standard#the-token-standard-field) `Fungible` |
...
?